tests/rest/client/test_room_access_rules.py

```diff
@@ -224,6 +229,25 @@ def test_public_room(self):
             self.current_rule_in_room(init_state_room_id), AccessRules.RESTRICTED
         )
 
+        # List the rooms in the public room list
+        request, channel = self.make_request(
+            "PUT",
+            "/_matrix/client/r0/directory/list/room/%s" % (preset_room_id,),
+            {"visibility": "public"},
+            access_token=self.tok,
+        )
+        self.render(request)
+        self.assertEqual(channel.code, 200, channel.result)
+
+        request, channel = self.make_request(
```

Comment:
This bit is easily mistaken for a duplicate of the one above, maybe add a comment explaining what it does.